core/sr-io/src/lib.rs

```diff
@@ -200,11 +200,41 @@ export_api! {
 
 export_api! {
 	pub(crate) trait CryptoApi {
-		/// Verify a ed25519 signature.
-		fn ed25519_verify<P: AsRef<[u8]>>(sig: &[u8; 64], msg: &[u8], pubkey: P) -> bool;
+		/// Generate an ed22519 key for the given key type and store it in the keystore.
+		///
+		/// Returns the raw public key.
+		fn ed25519_generate(id: KeyTypeId, seed: Option<&str>) -> [u8; 32];
```

Comment:
this is non-deterministic so it should certainly not be in the standard API. it should be offchain only.

Reply:
That is correct, we just do not set the keystore and then these functions are not usable from the runtime.